src/components/TradeDetails/TradeDetails.tsx

```diff
@@ -4,40 +4,48 @@ import { TokenInfo } from '@airswap/typescript';
 import { format } from '@greypixel_/nicenumbers';
 import { BigNumber } from 'bignumber.js';
 
+import { CollectionToken } from '../../entities/CollectionToken/CollectionToken';
+
 import './TradeDetails.scss';
 
 interface TradeDetailsProps {
   amount?: BigNumber;
   amountSubtext?: string;
   title: string;
-  tokenInfo: TokenInfo;
+  token: TokenInfo | CollectionToken;
```

Comment:
Why are we allowing both of these? We control the data that gets sent here? Should we not have cast it before it makes it here?

Reply:
OK that make's sense.
Maybe we could look to rename these separate to this PR? Whilst I now know and you know, I won't remember in a years time and (whilst I can't comment for your memory) somebody new wouldn't know that either. Doesn't need to be pretty `erc20TokenInfo` & `NftCollectionToken` would be waaay more explanatory for "the next person", though that's based on their use here, I don't know if that still makes sense elsewhere.